client/src/retry.rs

```diff
@@ -146,16 +146,26 @@ where
             throttle_backoff: throttle_cfg.into_exp_backoff(throttle_clock),
         }
     }
-    const fn should_log_retry_warning(&self, cur_attempt: usize) -> bool {
+
+    fn maybe_log_retry(&self, cur_attempt: usize, err: &tonic::Status) {
+        let mut do_log = false;
         // Warn on more than 5 retries for unlimited retrying
         if self.max_retries == 0 && cur_attempt > 5 {
-            return true;
+            do_log = true;
         }
         // Warn if the attempts are more than 50% of max retries
         if self.max_retries > 0 && cur_attempt * 2 >= self.max_retries {
-            return true;
+            do_log = true;
+        }
+
+        if do_log {
```

Comment:
At what speed can this repeat? Should we be worried about filling disk?

Reply:
It'll only go forever for long polling, which will hit the max retry interval of 10s quickly.

For others, they might retry faster but will eventually hit max retries.

In principle someone could specify a custom retry policy that just goes ham but, then they'd be absolutely blowing up their server too so I'm not too stressed.